### PR TITLE
Allow RSCT image to configure runtime via the RSCT spec

### PR DESCRIPTION
**Pull request by varad-ahirwadkar**
This PR fixes https://github.com/ocp-power-automation/rsct-operator/issues/13 and https://github.com/ocp-power-automation/rsct-operator/issues/12.

This PR allows the RSCT image runtime to be configured via the RSCT spec.

### CODE REVIEW

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: api/v1alpha1/rsct_types.go [hunk 1]
```diff
@@ -23,19 +23,25 @@ import (
...
 type RSCTSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of RSCT. Edit rsct_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Image is an RSCT image
+	// +optional
+	// +kubebuilder:default=`quay.io/powercloud/rsct-ppc64le:latest`
+	RSCTImage string `json:"image,omitempty"`
```
[the remark this answers is not in the document]
Comment (varad-ahirwadkar) [reply api/v1alpha1/rsct_types.go]: Ok, updated the code

File: api/v1alpha1/rsct_types.go [hunk 2]
```diff
@@ -23,19 +23,25 @@ import (
...
 // RSCTSpec defines the desired state of RSCT
 type RSCTSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of RSCT. Edit rsct_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Image is an RSCT image
+	// +optional
+	// +kubebuilder:default=`quay.io/powercloud/rsct-ppc64le:latest`
```
[the remark this answers is not in the document]
Comment (varad-ahirwadkar) [reply api/v1alpha1/rsct_types.go]: Using double quotes now

File: internal/controller/rsct_controller.go [hunk 3]
```diff
@@ -71,10 +71,15 @@ func (r *RSCTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return reconcile.Result{}, fmt.Errorf("failed to get RSCT %s: %w", req, err)
 	}
 
-	// TODO(mjturek): Allow image specification
 	r.Config.Namespace = rsct.Namespace
 	r.Config.Name = rsct.Name
-	r.Config.Image = "quay.io/powercloud/rsct-ppc64le:latest"
+
+	// Using the default RSCT image if not specified in RSCTSpec
+	if rsct.Spec.RSCTImage == "" {
```
[the remark this answers is not in the document]
Comment (varad-ahirwadkar) [reply internal/controller/rsct_controller.go]: Sure, updated

File: internal/controller/rsct_controller.go [hunk 4]
```diff
@@ -71,10 +71,15 @@ func (r *RSCTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return reconcile.Result{}, fmt.Errorf("failed to get RSCT %s: %w", req, err)
 	}
 
-	// TODO(mjturek): Allow image specification
 	r.Config.Namespace = rsct.Namespace
 	r.Config.Name = rsct.Name
-	r.Config.Image = "quay.io/powercloud/rsct-ppc64le:latest"
+
+	// Using the default RSCT image if not specified in RSCTSpec
+	if *rsct.Spec.Image == "" {
```
Comment (mkumatag) [inline internal/controller/rsct_controller.go:78]: check for nil check before checking for empty string

Comment (varad-ahirwadkar) [reply internal/controller/rsct_controller.go]: Added check for nil instead of empty string

Comment (varad-ahirwadkar) [reply internal/controller/rsct_controller.go]: Now, it is fetching default value. If the spec is empty, the default value will be used.
```
apiVersion: rsct.ibm.com/v1alpha1
kind: RSCT
metadata:
  name: rsct
  namespace: rsct-operator-system
spec: {}
```

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: internal/controller/rsct_controller.go [hunk 5]
```diff
@@ -71,10 +71,9 @@ func (r *RSCTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return reconcile.Result{}, fmt.Errorf("failed to get RSCT %s: %w", req, err)
 	}
 
-	// TODO(mjturek): Allow image specification
 	r.Config.Namespace = rsct.Namespace
 	r.Config.Name = rsct.Name
-	r.Config.Image = "quay.io/powercloud/rsct-ppc64le:latest"
+	r.Config.Image = *rsct.Spec.Image
```
[the remark this answers is not in the document]
Comment (varad-ahirwadkar) [reply internal/controller/rsct_controller.go]: ok, added nil check

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**